File: backend/src/main/java/com/go_exchange_easier/backend/dto/search/GetUserByExchangeResponse.java
package com.go_exchange_easier.backend.dto.search;

import java.time.LocalDate;

public record GetUserByExchangeResponse(

        Integer id,
        String nick,
        ExchangeDto exchange

) {

    public record ExchangeDto(

            Integer id,
            TimeRangeDto timeRange,
            UniversityDto university,
            UniversityMajorDto universityMajor

    ) {}

    public record TimeRangeDto(

            LocalDate startedAt,
            LocalDate endAt

    ) {}

    public record UniversityDto(

            Short id,
            String nativeName,
            String englishName,
            CityDto city

    ) {}

    public record UniversityMajorDto(

            Short id,
            String name

    ) {}

    public record CityDto(

            Integer id,
            String name,
            CountryDto country

    ) {}

    public record CountryDto(

            Short id,
            String name

    ) {}

}

File: frontend/src/dtos/search/GetUserByExchangeResponse.ts
import type { PageResponse } from "../common/PageResponse";

export type GetUserByExchangeResponse = PageResponse<{
  id: number;
  nick: string;
  exchange: {
    id: null;
    timeRange: {
      startedAt: string;
      endAt: string;
    };
    university: {
      id: number;
      nativeName: string;
      englishName: string | null;
      city: {
        id: number;
        name: string;
        country: {
          id: number;
          name: string;
        };
      };
    };
    universityMajor: {
      id: number;
      name: string;
    };
  };
}>;
